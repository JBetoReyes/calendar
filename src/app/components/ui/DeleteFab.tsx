/* eslint-disable @typescript-eslint/indent */
/* eslint-disable @typescript-eslint/no-explicit-any */
import {faTrash} from '@fortawesome/free-solid-svg-icons';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import React from 'react';
import {connect} from 'react-redux';
import {IStoreState} from '../../store/storeModel';
import {AppClickEvent} from '../../../typings/htmlEvents';
import {startDeleteEvent} from '../../reducers/calendarActions';

const mapDispatchToProps = {
  startDeleteEvent,
};
const mapStateToProps = (state: IStoreState) => ({
  activeEvent: state.calendar.activeEvent,
});
type DispatchPropsType = typeof mapDispatchToProps;
type MapStateToPropsType = ReturnType<typeof mapStateToProps>;
type OwnPropsType = Record<string, any>;
type PropsType = DispatchPropsType & MapStateToPropsType;
const DeleteFab = (props: OwnPropsType): JSX.Element => {
  const {startDeleteEvent: dispatchDeleteEvent} = props as PropsType;
  const handleDeleteEvent = (e: AppClickEvent) => {
    dispatchDeleteEvent();
  };
  return (
    <button
      type="button"
      className="btn btn-danger position-fixed fixed-bottom m-5"
      onClick={handleDeleteEvent}
    >
      <FontAwesomeIcon icon={faTrash} size="1x" /> Delete
    </button>
  );
};

export default connect<
  MapStateToPropsType,
  DispatchPropsType,
  OwnPropsType,
  IStoreState
>(
  mapStateToProps,
  mapDispatchToProps,
)(DeleteFab);
