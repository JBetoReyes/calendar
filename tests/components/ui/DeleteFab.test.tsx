import React, {ComponentType} from 'react';
import {render, mount, shallow} from 'enzyme';
import {Provider} from 'react-redux';
import configureStore, {MockStoreEnhanced} from 'redux-mock-store';
import thunk from 'redux-thunk';
import DeleteFab from '../../../src/app/components/ui/DeleteFab';

type RenderType = typeof render | typeof mount | typeof shallow;
const setup = (
  Component: ComponentType,
  props: Partial<Record<string, unknown>> = {},
  renderType: RenderType = mount,
) => {
  return (renderType as CallableFunction)(<Component {...props} />);
};

const middlewares = [thunk];
const mockStoreProvider = configureStore(middlewares);
let store: MockStoreEnhanced;
describe('DeleteFab', () => {
  test('Should match with snapshot', () => {
    store = mockStoreProvider({
      calendar: {
        activeEvent: {},
      },
    });
    const component = () => (
      <Provider store={store}>
        <DeleteFab />
      </Provider>
    );
    const wrapper = setup(component);
    expect(wrapper).toMatchSnapshot();
  });
});
