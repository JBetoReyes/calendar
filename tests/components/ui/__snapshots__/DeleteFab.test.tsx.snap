// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFab Should match with snapshot 1`] = `
<component>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(DeleteFab)>
      <DeleteFab
        activeEvent={Object {}}
        startDeleteEvent={[Function]}
      >
        <button
          className="btn btn-danger position-fixed fixed-bottom m-5"
          onClick={[Function]}
          type="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f1f8",
                  "M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z",
                ],
                "iconName": "trash",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size="1x"
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-trash fa-w-14 fa-1x "
              data-icon="trash"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
           Delete
        </button>
      </DeleteFab>
    </Connect(DeleteFab)>
  </Provider>
</component>
`;
